frontend/public/components/utils/horizontal-nav.tsx

```diff
@@ -276,22 +281,33 @@ export const HorizontalNav = React.memo((props: HorizontalNavProps) => {
     [navTabExtensions, objReference],
   );
 
-  const dynamicPluginPages = React.useMemo(
-    () =>
-      navTabExtentionsResolved
-        ? dynamicNavTabExtensions
-            .filter(
-              (tab) =>
-                referenceForExtensionModel(tab.properties.model as ExtensionK8sGroupModel) ===
-                objReference,
-            )
-            .map((tab) => ({
-              ...tab.properties.page,
-              component: tab.properties.component,
-            }))
-        : [],
-    [dynamicNavTabExtensions, navTabExtentionsResolved, objReference],
-  );
+  const dynamicPluginPages = React.useMemo(() => {
+    const resolvedResourceNavTab = dynamicResourceNavTabExtensions
+      .filter(
+        (tab) =>
+          referenceForExtensionModel(tab.properties.model as ExtensionK8sGroupModel) ===
+          objReference,
+      )
+      .map((tab) => ({
+        ...tab.properties.page,
+        component: (params: PageComponentProps) => (
+          <AsyncComponent {...params} loader={tab.properties.component} />
+        ),
+      }));
+
+    const resolvedNavTab = dynamicTabExtensions
+      .filter((tab) => tab.properties.contextId === contextId)
+      .map((tab) => ({
+        ...tab.properties,
```

Comment:
You only need those properties that actually are relevant to be used later. ie `name` and `href`